src/common/docker-compose/index.ts

```diff
@@ -54,6 +37,31 @@ export const generate = (dependency_manager: DependencyManager): DockerComposeTe
           ...node.parameters,
         },
       };
+
+      for (const [param_name, param_value] of Object.entries(compose.services[node.normalized_ref].environment || {})) {
+        env_params_to_expand[`${node.normalized_ref.toUpperCase()}_${param_name}`.replace(/[.-]/g, '_')] = param_value;
+      }
+      for (const [param_name, param_value] of Object.entries(node.service_config.getParameters())) {
```

Comment:
The logic inside this iterator should go into the `getParamValues()` method on the dependency manager. We already parse the service configs there and enrich the key/value parameters on the nodes, so we should be sure to consolidate that logic.